parser/Builder.cc

```diff
@@ -1727,6 +1752,22 @@ class Builder::Impl {
                parser::isa_node<Hash>(&node);
     }
 
+    void checkAssignmentToNumberedParameters(std::string_view name, core::LocOffsets loc) {
+        if (driver_->lex.context.inDynamicBlock() && isNumberedParameterName(name) &&
+            driver_->numparam_stack.seen_numparams()) {
+            std::cout << "Assignment error" << std::endl;
+            core::Loc location = core::Loc(file_, loc);
+            if (auto e = gs_.beginError(location, core::errors::Parser::AssignmentToNumparamError)) {
+                e.setHeader("cannot assign to numbered parameter {}", name);
+                e.addErrorNote("Reserved numbered parameter names are not allowed starting with Ruby 3.0. Use {} to "
```

Comment:
```suggestion
                e.addErrorNote("Reserved numbered parameter names are not allowed starting with Ruby 3.0. Use `{}` to "
```

using

```
`{}`
```

instead of just `{}` will cause the interpolated string to be colored